numpy/core/src/common/array_assign.c

```diff
@@ -84,14 +84,27 @@ broadcast_error: {
 
 /* See array_assign.h for parameter documentation */
 NPY_NO_EXPORT int
-raw_array_is_aligned(int ndim, char *data, npy_intp *strides, int alignment)
+raw_array_is_aligned(int ndim, npy_intp *shape,
+                     char *data, npy_intp *strides, int alignment)
 {
     if (alignment > 1) {
         npy_intp align_check = (npy_intp)data;
```

Comment:
Maybe `npy_uintp`. I'm pretty sure both work here for twos complement integers, but I needed to think about it.

Reply:
Note that it is later cast to `npy_uintp` in `npy_is_aligned`.

Reply:
Only bitwise operations are involved, so it should be exactly the same.